client/js/components/display/index.js

```diff
@@ -2,28 +2,52 @@ const html = require('choo/html')
 const fourohfour = require('../../elements/404')
 
 // XXX: server-side data rendering could pull from a cache if we want
-const data = module.parent ? function () { } : require('render-data')
+const renderData = module.parent ? function () { } : require('render-data')
 const display = html`<div id="item"></div>`
 
 module.exports = function (state, prev, send) {
-  const entryName = state.preview.entryName
   const archive = state.archive.instance
+  const entryName = state.preview.entryName
+  const previousEntryName = prev && prev.preview ? prev.preview.entryName : null
+
+  // only render/re-render when the entry name changes!
+  if (!entryName || (entryName === previousEntryName)) {
```

Comment:
@karissa: this is important: this is what was causing the preview display seizures -- redux-style frameworks like choo call all their reducers and trigger re-renders of all components when anything mutates the models they are based on, so its important to have a gatekeeping check like the one above, otherwise every time the state changes in any of the models, all hell breaks loose in components.